homeassistant/components/statistics/sensor.py

```diff
@@ -499,10 +500,19 @@ async def _initialize_from_database(self) -> None:
             else:
                 _LOGGER.debug("%s: retrieving all records", self.entity_id)
 
+            query = query.outerjoin(
+                StateAttributes, States.attributes_id == StateAttributes.attributes_id
+            )
             query = query.order_by(States.last_updated.desc()).limit(
                 self._samples_max_buffer_size
             )
-            states = execute(query, to_native=True, validate_entity_ids=False)
+            if results := execute(query, to_native=False, validate_entity_ids=False):
+                # After 2023.8 make state.to_native require StateAttributes
```

Comment:
This is doing I/O in the event loop but its existing so I opened https://github.com/home-assistant/core/issues/68282